Tests/TuistKitTests/Services/TestServiceTests.swift

```diff
@@ -19,17 +19,20 @@ final class TestServiceTests: TuistUnitTestCase {
     private var xcodebuildController: MockXcodeBuildController!
     private var buildGraphInspector: MockBuildGraphInspector!
     private var simulatorController: MockSimulatorController!
+    private var testsCacheTemporaryDirectory: TemporaryDirectory!
 
     override func setUpWithError() throws {
         try super.setUpWithError()
         generator = .init()
         xcodebuildController = .init()
         buildGraphInspector = .init()
         simulatorController = .init()
+        testsCacheTemporaryDirectory = try TemporaryDirectory(removeTreeOnDeinit: true)
```

Comment:
You'll have to update the tearDown to set this one to nil. Otherwise, it'll remain allocated in memory until the completion of the entire test suite.